demos/projects/ESPRESSIF/adu/components/sample-azure-iot/transport_tls_esp32.c

```diff
@@ -134,6 +134,11 @@ TlsTransportStatus_t TLS_Socket_Connect( NetworkContext_t * pNetworkContext,
         esp_transport_ssl_set_client_key_data_der( pxEspTlsTransport->xTransport, (const char *) pNetworkCredentials->pucPrivateKey, pNetworkCredentials->xPrivateKeySize );
     }
 
+	if( pxEspTlsTransport == NULL )
```

Comment:
There is already a NULL check for this variable beforehand in line 95. What is different in this case? Have you run into the null ptr dereference error?

Reply:
Then the test in line 155 is redundant too

Reply:
Unless there is a call in between 95 and 137 that can change `pxEspTlsTransport` to NULL (from what I see there are only calls setting variables in `pxEspTlsTransport` not the variable itself), the check you've added would not be hit. If that is not the case, could you point me to which function?